### ISSUE TITLE
Unable to parse imports

### ISSUE THREAD

**Issue by DrJedikiah** [mactec0/Kernelmode-manual-mapping-through-IAT#3]
It works with regular apps in kernelmode injection but while it's still kernelmode, it gives that error with EAC enabled game.

**Comment by DrJedikiah** [comment]
![Screenshot_2](https://user-images.githubusercontent.com/12988451/62664053-b9774800-b982-11e9-8d26-45b7bb73b3c3.png)

This is how it looks in debug mode. There are two imports but they don't have a regular name. Does it have to be specific Translate one or can I use one of them?

**Comment by mactec0** [comment]
Check if the imports inside the target process are stored correctly. This game might be packed with some kind of protector(e.g. VMProtect) which gives ability to protect imports by storing them differently.

I've used TranslateMessage, because it's often used by many programs and it's called very frequently. With minor changes, you can use a different function.

**Comment by Fatulatti** [comment]
I've tried using this on Rainbow Six, and it shows around 20 imports when debugging, none of them are TranslateMessage though.
Game uses 100% TranslateMessage though.

**Comment by DrJedikiah** [comment]
Yes, looks like it's packed. Do you recommend a way to find the original IAT?

**Comment by DrJedikiah** [comment]
What if I get a signature from translatemessage and search in the memory? I'm gonna try that.

Okay it found the TranslateMessage function but I guess I still need the IAT pointer that points this function's address right?

**Comment by mactec0** [comment]
You can stop AC service, attach debugger and break inside TranslateMessage to find out how it's being called. 
Though in this case it would probably be better to use some other place to the make the hook(e.g. vmt table entry, inline code hook), because tampering with packer protection will probably be harder to overcome.

**Comment by mactec0** [comment]
If you are still having this issue here is an example of using vmt table:
https://github.com/mactec0/Kernelmode-manual-mapping-through-IAT/tree/swapchain_vmt_example

**Comment by DrJedikiah** [comment]
Thank you, I'll try it.
